websites/D/Disney+/presence.ts

```diff
@@ -37,85 +55,149 @@ presence.on("UpdateData", async () => {
+            url: `https://www.disneyplus.com/groupwatch/${groupWatchId}`
+          });
+        }
+      }
+
       if (title) presence.setActivity(data, !video.paused);
     }
-    // Disney+ Hotstar video
+
+  // GroupWatch lobby
+  } else if (isHostDP && location.pathname.includes("/groupwatch/")) {
+    groupWatchCount = document.querySelectorAll(
+      ".gw-avatar-enter-done:not([id=gw-invite-button])"
+    ).length;
+
+    const seriesFields: NodeListOf<HTMLDivElement> = document.querySelectorAll(`
+      #webAppScene main #group + div:not([id]) h3[style]:nth-of-type(1),
+      #webAppScene main #group + div:not([id]) h3[style]:nth-of-type(2)
+    `);
+
+    if (seriesFields.length > 0) {
+      title = seriesFields[0]?.textContent;
+      subtitle = seriesFields[1]?.textContent;
+    } else {
+      const movieField: HTMLImageElement = document.querySelector(
+        "#webAppScene main #group + div:not([id]) img[alt]"
+      );
+      title = movieField?.alt;
+    }
+
+    data.details = `${title} ${subtitle ? `- ${subtitle}` : ""}`;
+    data.state = "Starting a GroupWatch";
+    // set GroupWatch participants size
+    data.partySize = groupWatchCount;
+    data.partyMax = 7;
+
+    // add button, if enabled
+    if (buttons && groupWatchBtn) {
+      data.buttons = [{
+        label: "Join GroupWatch",
+        url: location.pathname
+      }];
+    }
+
+    if (title) presence.setActivity(data, false);
+
+  // Disney+ Hotstar video
   } else if (isHostHS && /\/(tv|movies)\//.test(location.pathname)) {
     const video: HTMLVideoElement = document.querySelector(
       ".player-base video"
     );
 
     if (video && !isNaN(video.duration)) {
-      const timestamps: number[] = getTimestamps(
-        Math.floor(video.currentTime),
-        Math.floor(video.duration)
-      );
+      const timestamps: number[] = presence.getTimestampsfromMedia(video),
+            titleField: HTMLDivElement = document.querySelector(".controls-overlay .primary-title"),
+            subtitleField: HTMLDivElement = document.querySelector(".controls-overlay .show-title");
 
-      const titleField: HTMLDivElement = document.querySelector(
-        ".controls-overlay .primary-title"
-      );
-      title = titleField ? titleField.textContent : null;
-      const subtitleField: HTMLDivElement = document.querySelector(
-        ".controls-overlay .show-title"
-      );
-      subtitle = subtitleField ? subtitleField.textContent : null;
+      title = titleField?.textContent;
+      subtitle = subtitleField?.textContent; // episode or empty if it's a movie
 
-      // subtitleField is episode for series, empty for movies
       data.details = title;
-      data.state = subtitle ? subtitle : "Movie";
-
+      data.state = subtitle || "Movie";
       data.smallImageKey = video.paused ? "pause" : "play";
       data.smallImageText = video.paused
-        ? (await strings).pause
-        : (await strings).play;
+        ? strings.pause
+        : strings.play;
```

Comment:
do await for sure...

Reply:
`await` is already used on L48 and therefore not needed any further, but I could change it back if that's preferred.